webapp/src/main/java/com/box/l10n/mojito/smartling/SmartlingClient.java

```diff
@@ -0,0 +1,199 @@
+    Integer authenticationDurationHours = 23;
+
+    public SmartlingClient(String userIdentifier, String userSecret) {
+        this.userIdentifier = userIdentifier;
+        this.userSecret = userSecret;
+    }
+
+    public enum RequestType {
+        AUTHENTICATION,
+        REFRESH,
+        SOURCE_STRINGS,
+        WITH_TOKEN
+    }
+
+    Map<String, Object> getPayloadMap(RequestType requestType) {
+        Map<String, Object> map = new HashMap<>();
+        if (requestType == RequestType.AUTHENTICATION) {
+            map.put("userIdentifier", userIdentifier);
+            map.put("userSecret", userSecret);
+        } else if (requestType == RequestType.REFRESH) {
+            map.put("refreshToken", refreshToken);
+        }
+        return map;
+    }
+
+    HttpHeaders getHttpHeadersForJson(RequestType requestType) {
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        if (requestType == RequestType.SOURCE_STRINGS) {
+            headers.set("Authorization", "Bearer " + authToken);
+        }
+        return headers;
+    }
+
+    <T> HttpEntity<T> getHttpEntityForPayload(T payload, RequestType requestType) {
+        return new HttpEntity<>(payload, getHttpHeadersForJson(requestType));
+    }
+
+    public AuthenticationResponse authenticate() throws SmartlingClientException {
+        Map<String, Object> payload = getPayloadMap(RequestType.AUTHENTICATION);
+
+        HttpEntity<Map<String, Object>> httpEntity = getHttpEntityForPayload(payload, RequestType.AUTHENTICATION);
+        AuthenticationResponse authenticationResponse;
+        try {
+            authenticationResponse = postForAuthentication(httpEntity, AUTH_API_V2_AUTHENTICATE);
+        } catch (SmartlingClientException e) {
+            throw new SmartlingClientException(e.getMessage());
+        }
+        if (!authenticationResponse.getResponse().getCode().equals("SUCCESS")) {
+            String msg = formatAuthenticationErrorMessage(authenticationResponse);
+            throw new SmartlingClientException(msg);
+        } else {
+            Instant now = Instant.now();
+            lastAuthentication = now;
+            lastRefresh = now;
+            nextAuthentication = now.plus(Duration.ofHours(authenticationDurationHours));
+            nextRefresh = now.plus(Duration.ofMinutes(refreshDurationMinutes));
+            authToken = authenticationResponse.getResponse().getData().getAccessToken();
+            refreshToken = authenticationResponse.getResponse().getData().getRefreshToken();
+        }
+        return authenticationResponse;
+    }
+
+    AuthenticationResponse postForAuthentication(HttpEntity<Map<String, Object>> httpEntity, String authApiV2) throws SmartlingClientException {
+        String postForObject = restTemplate.postForObject(
+                authApiV2, httpEntity, String.class);
+        ObjectMapper json = new ObjectMapper();
```

Comment:
no need to re-instanciate the `ObjectMapper` all the time, should be moved as an attribute. there is shared one in the app that can be autowired as well